File: benchmarks/struct_array_push.js
const n = 1_000_000;
const points = [];
for (let i = 0; i < n; i++) {
    points.push({ x: i, y: i * 2, z: i * 3 });
}
let total = 0;
for (const p of points) {
    total += p.x + p.y + p.z;
}
console.log(total);

File: benchmarks/struct_array_push.rs
struct V3 {
    x: i64,
    y: i64,
    z: i64,
}

fn main() {
    let n = 1_000_000;
    let mut points = Vec::with_capacity(n);
    for i in 0..n as i64 {
        points.push(V3 { x: i, y: i * 2, z: i * 3 });
    }
    let mut total: i64 = 0;
    for p in &points {
        total += p.x + p.y + p.z;
    }
    println!("{}", total);
}
